transport_plugins/bled112/iotile_transport_bled112/bled112.py

```diff
@@ -569,21 +579,40 @@ def _process_scan_event(self, response):
 
             if data[22] == 0xFF and data[23] == 0xC0 and data[24] == 0x3:
                 self._v1_scan_count += 1
-                self._parse_v1_advertisement(rssi, string_address, data)
+                info = self._parse_v1_advertisement(rssi, string_address, data)
             elif data[3] == 27 and data[4] == 0x16 and data[5] == 0xdd and data[6] == 0xfd:
                 self._v2_scan_count += 1
-                self._parse_v2_advertisement(rssi, string_address, data)
+                info, reading_time, stream, reading, \
+                    broadcast_toggle, counter, broadcast_multiplex = \
+                    self._parse_v2_advertisement(rssi, string_address, data)
```

Comment:
This is going to fail when you return as you do on line 615. There, you specified 4 Nones, but you are unpacking 7 values. Example:
```
>>> def f(a=0):
...   if a == 0:
...     return None, None, None
...   return 1, 2, 3, 4, 5
... 
>>> a,b,c,d,e = f()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: not enough values to unpack (expected 5, got 3)
```

Reply:
Thank you for pointing out, this I missed!